src/main/java/htsjdk/samtools/util/AbstractRecordAndOffset.java

```diff
@@ -0,0 +1,132 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package htsjdk.samtools.util;
+
+import htsjdk.samtools.SAMRecord;
+
+/**
+ * Holds a SAMRecord plus the zero-based offset into that SAMRecord's bases and quality scores that corresponds
+ * to the base and quality at the genomic position described the containing AbstractLocusInfo. One object represents
+ * one base for <code>SamLocusIterator.RecordAndOffset</code> implementation or one alignment block of
+ * <code>SAMRecord</code> for <code>TypedRecordAndOffset</code> implementation.
+ * 
+ * @author Darina_Nikolaeva@epam.com, EPAM Systems, Inc. <www.epam.com>
+ * 
+ */
+public class AbstractRecordAndOffset {
+
+    /**
+     * A SAMRecord aligned to reference position
+     */
+    protected final SAMRecord record;
+    /**
+     * Zero-based offset into the read corresponding to the current position in AbstractLocusInfo
+     */
+    protected final int offset;
+
+    /**
+     * @param record inner SAMRecord
+     * @param offset from the start of the read
+     * @param length of alignment block
+     * @param refPos corresponding to read offset reference position
+     */
+    public AbstractRecordAndOffset(final SAMRecord record, final int offset, int length, int refPos) {
+        this(record, offset);
+    }
+
+    /**
+     * @param record inner SAMRecord
+     * @param offset from the start of the read
+     */
+    public AbstractRecordAndOffset(final SAMRecord record, final int offset) {
+        this.offset = offset;
+        this.record = record;
+    }
+
+    /**
+     * @return offset of aligned read base from the start of the read.
+     */
+    public int getOffset() {
+        return offset;
+    }
+
+    /**
+     * @return inner <code>SAMRecord</code> object.
+     */
+    public SAMRecord getRecord() {
+        return record;
+    }
+
+    /**
+     * @return the read base according to <code>offset</code>.
+     */
+    public byte getReadBase() {
+        return record.getReadBases()[offset];
+    }
+
+    /**
+     * @return the length of alignment block represented by the object.
+     */
+    public int getLength() {
+        return 1;
+    }
+
+    /**
+     * @return the position in reference sequence, to which the start of alignment block is aligned.
+     */
+    public int getRefPos() {
+        return 0;
+    }
+
+    /**
+     * @return read name of inner SAMRecord.
+     */
+    public String getReadName() {
+        return record.getReadName();
+    }
+
+    /**
+     * @return array of base qualities of inner SAMRecord.
+     */
+    public byte[] getBaseQualities() {
+        return record.getBaseQualities();
+    }
+    
+    /**
+     * @return the base quality according to <code>offset</code>.
+     */
+    public byte getBaseQuality() {
+        return record.getBaseQualities()[offset];
+    }
+
+    /**
+     * @param position in the reference
```

Comment:
I still do not understand the comment here. if `position` is a number in the reference (i.e. can be as big as the length of the reference), and offset is a size smaller than the length of the read then how can you access record.getBaseQualities()[position - offset] ? I think that position must be a position in the read, not the reference.